Kitodo/src/main/java/org/kitodo/production/services/index/ServerConnectionChecker.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.services.index;
+
+import static java.lang.Thread.currentThread;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.InvalidPropertiesFormatException;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kitodo.config.KitodoConfig;
+
+/**
+ * Checks the connection to the search service. This is an asynchronous function
+ * to {@link IndexingService}. It uses its variables for inter-thread
+ * communication.
+ */
+class ServerConnectionChecker implements Runnable {
+    private static final Logger logger = LogManager.getLogger(ServerConnectionChecker.class);
+    private static final Pattern PATTERN_SERVER = Pattern.compile("cluster_name\\W+([^\"]*).*?number\\W+([^\"]*)",
```

Comment:
Is this a pattern which is working with OpenSearch and ElasticSearch servers?

Reply:
I don't know yet, right now I'm focused on getting the existing code to work. If not, it could be extended. I assume both are the same, but I haven't tried it yet. First, it's focused on finishing the minimal development.

Reply:
Ok, fine for me and hopefully it is working without a change.